From 259d92fa848d296e9e4e2c43153ba2d9eeac10fd Mon Sep 17 00:00:00 2001
From: Etienne Carriere <etienne.carriere@foss.st.com>
Date: Wed, 27 Sep 2023 17:49:29 +0200
Subject: [PATCH 0675/1695] dts: stm32: add SAES and PKA dependency on RNG
 clock for stm32mp13

Adds missing RNG clock resource in SAES and PKA nodes in stm32mp13
SoC DTSI files.

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/optee_os.git 84778c9830deb3067015f5b2ddd4aefa157663d2 dts: stm32: add SAES and PKA dependency on RNG clock for stm32mp13]
Change-Id: I6fbf6c0d356d81b9c6e868faf80c6a282c98d8c7
Signed-off-by: Etienne Carriere <etienne.carriere@foss.st.com>
Reviewed-on: https://gerrit.st.com/c/mpu/oe/optee/optee_os/+/330038
Reviewed-on: https://gerrit.st.com/c/mpu/oe/optee/optee_os/+/361858
Tested-by: Thomas BOURGOIN <thomas.bourgoin@foss.st.com>
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
Reviewed-by: Gatien CHEVALLIER <gatien.chevallier@st.com>
Reviewed-by: Thomas BOURGOIN <thomas.bourgoin@foss.st.com>
Domain-Review: Gatien CHEVALLIER <gatien.chevallier@st.com>
---
 core/arch/arm/dts/stm32mp13xc.dtsi | 6 ++++--
 core/arch/arm/dts/stm32mp13xf.dtsi | 6 ++++--
 2 files changed, 8 insertions(+), 4 deletions(-)

diff --git a/core/arch/arm/dts/stm32mp13xc.dtsi b/core/arch/arm/dts/stm32mp13xc.dtsi
index 51bab0c0d..9fb961444 100644
--- a/core/arch/arm/dts/stm32mp13xc.dtsi
+++ b/core/arch/arm/dts/stm32mp13xc.dtsi
@@ -19,7 +19,8 @@
 		compatible = "st,stm32mp13-saes";
 		reg = <0x54005000 0x400>;
 		interrupts = <GIC_SPI 82 IRQ_TYPE_LEVEL_HIGH>;
-		clocks = <&rcc SAES_K>;
+		clocks = <&rcc SAES_K>, <&rcc RNG1_K>;
+		clock-names = "bus", "rng";
 		resets = <&rcc SAES_R>;
 		status = "disabled";
 	};
@@ -28,7 +29,8 @@
 		compatible = "st,stm32mp13-pka64";
 		reg = <0x54006000 0x2000>;
 		interrupts = <GIC_SPI 105 IRQ_TYPE_LEVEL_HIGH>;
-		clocks = <&rcc PKA>;
+		clocks = <&rcc PKA>, <&rcc RNG1_K>;
+		clock-names = "bus", "rng";
 		resets = <&rcc PKA_R>;
 		status = "disabled";
 	};
diff --git a/core/arch/arm/dts/stm32mp13xf.dtsi b/core/arch/arm/dts/stm32mp13xf.dtsi
index dd106b50f..39a179352 100644
--- a/core/arch/arm/dts/stm32mp13xf.dtsi
+++ b/core/arch/arm/dts/stm32mp13xf.dtsi
@@ -19,7 +19,8 @@
 		compatible = "st,stm32mp13-saes";
 		reg = <0x54005000 0x400>;
 		interrupts = <GIC_SPI 82 IRQ_TYPE_LEVEL_HIGH>;
-		clocks = <&rcc SAES_K>;
+		clocks = <&rcc SAES_K>, <&rcc RNG1_K>;
+		clock-names = "bus", "rng";
 		resets = <&rcc SAES_R>;
 		status = "disabled";
 	};
@@ -28,7 +29,8 @@
 		compatible = "st,stm32mp13-pka64";
 		reg = <0x54006000 0x2000>;
 		interrupts = <GIC_SPI 105 IRQ_TYPE_LEVEL_HIGH>;
-		clocks = <&rcc PKA>;
+		clocks = <&rcc PKA>, <&rcc RNG1_K>;
+		clock-names = "bus", "rng";
 		resets = <&rcc PKA_R>;
 		status = "disabled";
 	};
-- 
2.47.2

